snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb

```diff
@@ -75,4 +75,30 @@ def launch_simultaneously(devices, language, locale, launch_arguments)
       return Collector.fetch_screenshots(raw_output, dir_name, '', launch_arguments.first)
     end
   end
+
+  class CPUInspector
+    def self.hwprefs_available?
+      `which hwprefs` != ''
+    end
+
+    def self.cpu_count
+      @cpu_count ||=
+        case RUBY_PLATFORM
+        when /darwin9/
+          `hwprefs cpu_count`.to_i
+        when /darwin10/
+          (hwprefs_available? ? `hwprefs thread_count` : `sysctl -n hw.physicalcpu_max`).to_i
+        when /linux/
```

Comment:
We could `UI.user_error!` if it's not macOS at this point (if we don't already to that at _snapshot_ somewhere)